extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts

```diff
@@ -234,7 +235,8 @@ export class ClusterSettingsPage extends WizardPageBase<DeployClusterWizard> {
 					self.wizard.registerDisposable(disposable);
 				},
 				onNewInputComponentCreated: (name: string, inputComponentInfo: InputComponentInfo): void => {
-					this.inputComponents[name] = { component: inputComponentInfo.component };
+					this.inputComponents[name] = inputComponentInfo;
```

Comment:
Hm, are you sure this is working? Passing in functions like this generally is going to have the wrong "this" context - hence capturing it in self.

Reply:
I will move the function out so that it is not tied to the 'this' context at all. There is no need for it to be an instance method.

Reply:
done.

Reply:
Yeah that's true - I always get those mixed up. Lambda functions shouldn't require using self